### ISSUE TITLE
Missing header file logger.hpp

### ISSUE THREAD

**Issue by tillea** [mateidavid/nanocall#9]
Hi,

I'm trying to create Debian packages of nanocall.  Unfortunately there is either a missing file logger.hpp or you might refer to some other logger.hpp (may be from libboost?) but the include options are not properly set.  In any case I get:

...
[  6%] Building CXX object nanocall/CMakeFiles/run-fwbw.dir/run-fwbw.cpp.o
cd /build/nanocall-0.6.14/obj-x86_64-linux-gnu/nanocall && /usr/bin/c++    -isystem /usr/include/hdf5/serial -I/build/nanocall-0.6.14/obj-x86_64-linux-gnu -I/build/nanocall-0.6.14/src/builtin_models -I/build/na
In file included from /build/nanocall-0.6.14/src/nanocall/Pore_Model.hpp:20:0,
                 from /build/nanocall-0.6.14/src/nanocall/run-fwbw.cpp:5:
/build/nanocall-0.6.14/src/nanocall/Event.hpp:10:22: fatal error: logger.hpp: No such file or directory
 #include "logger.hpp"
                      ^
compilation terminated.
...

Any hint how to get this header file?

Kind regards

```
    Andreas.
```

**Comment by mateidavid** [comment]
Hi.

The file is from `hpptools`, which is a submodule. So make sure you use `--recursive` with `git clone`.

Also- Why are you compiling `run-fwbw`? That target is only part of the Test and Debug build types, not of Release, which is the default. Did you explicitly set `CMAKE_BUILD_TYPE=Test`?

**Comment by tillea** [comment]
Hi

On Tue, Sep 13, 2016 at 07:17:02AM -0700, Matei David wrote:
> The file is from `hpptools`, which is a submodule. So make sure you use `--recursive` with `git clone`.

Thanks for the quick response.  The solution does not really help here
but I know what to do: Packaging hpptools as a separate Debian package.
 
> Also- Why are you compiling `run-fwbw`? That target is only part of the Test and Debug build types, not of Release, which is the default. Did you explicitly set `CMAKE_BUILD_TYPE=Test`?

Debian packaging defaults are also turning on tests which is IMHO
sensible behaviour to test what we ship.

I'll come back in case of further issues.

BTW, the src/Makefile does not work propely.  If you call

    make -f src/Makefile clean

it cleans the source directory rather than the executable.  I'd
recommend to remove this.

Kind regards

     Andreas.

-- 
http://fam-tille.de

**Comment by mateidavid** [comment]
A few points:

- I'm not sure if having `hpptools` headers installed in system-wide folders is a good idea. When I wrote it, I didn't have that use case in mind, I was only aiming for using it exactly as it is being used in Nanocall- as a submodule of another package. The problem might be that the file names I'm using are very short, e.g. `alg.hpp`, which might lead to conflicts. Perhaps putting those files in a separate folder would be a better idea: `/usr/include/hpptools/alg.hpp`, instead of `/usr/include/alg.hpp`. Maybe we should discuss this in an issue for that repo?

- Regarding the Nanocall Test build type- the files that you end up building, such as `run-fwbw` are really meant for development only. I don't think it's a good idea to have those part of a widely-distributed package. Unfortunately, I don't have simple test cases for Nanocall, and using build type Test instead of Release is not going to change that fact (there are no tests in build type Test either). Perhaps I could add some simple tests to the Release build to solve this issue. The problem is that I would need to make some Fast5 files public, I have to check if that's ok.

- Likewise, `src/Makefile` was used for early development, predating the CMake build system. By now, it's completely obsolete, I hadn't even noticed it. I'll remove that file to avoid any further confusion.

- I've been very liberal with tagging releases, especially with the third component. Essentially the only things I haven't tagged are meta-things, such as the Travis CI config file. Is that a problem if Nanocall is built into a package? If so, are there more specific guidelines of what is worth tagging?

**Comment by tillea** [comment]
On Tue, Sep 13, 2016 at 11:35:23AM -0700, Matei David wrote:
> would be a better idea: `/usr/include/hpptools/alg.hpp`, instead of `/usr/include/alg.hpp`. Maybe we should discuss this in an issue for that repo?

Its definitely the plan to put the header files into
/usr/include/hpptools.  Otherwise names would be too generic.
To compile Nanocall a -I/usr/include/hpptools/ would be
needed in CFLAGS.

> - Regarding the Nanocall Test build type- the files that you end up building, such as `run-fwbw` are really meant for development only. I don't think it's a good idea to have those part of a widely-distributed package. Unfortunately, I don't have simple test cases for Nanocall, and using build type Test instead of Release is not going to change that fact (there are no tests in build type Test either). Perhaps I could add some simple tests to the Release build to solve this issue. The problem is that I would need to make some Fast5 files public, I have to check if that's ok.

Thanks for the hint.  There is a difference between what is build in the
build process and what is distributed inside the installed package.  I
need to check out the build targets to get a hint what is user oriented
(to tired now).
 
> - Likewise, `src/Makefile` was used for early development, predating the CMake build system. By now, it's completely obsolete, I hadn't even noticed it. I'll remove that file to avoid any further confusion.

Yes, removing seems a good idea.
 
> - I've been very liberal with tagging releases, especially with the third component. Essentially the only things I haven't tagged are meta-things, such as the Travis CI config file. Is that a problem if Nanocall is built into a package? If so, are there more specific guidelines of what is worth tagging?

I think it is a good idea to tag what is tested and users should install
on their machine (no matter via a distribution or if users do manual
installation).
 
Kind regards

      Andreas.

-- 
http://fam-tille.de

**Comment by mateidavid** [comment]
I went back to this issue. I tagged new versions of `hpptools` and `fast5`. Then I pulled the updates into `nanocall`, and tagged the result as well. (I also made some more consistent changes, that's why I increases the second digit.) Try the new version to see if you find any more build problems.

`tclap` remains untagged, but since it is a header-only dependency (just like the others), it is not necessary to run Nanocall. Reproducibility is ensured by the fact that the Nanocall git repo contains a reference to a specific commit inside `tclap`.
